### PR TITLE
fix up tests so we ensure issue and prove are working

### PR DESCRIPTION
**Pull request by jsaur**
Adds some awaits, and updates some of the tests so we're sure we're getting credentials and proofs coming through

### CODE REVIEW

File: src/controller/agent.controller.controller.ts [hunk 1]
```diff
@@ -24,8 +24,9 @@ export class AgentControllerController {
     ): Promise<any> {
         Logger.log('----- GET WEB HOOK START -----');
         Logger.log(`${agentId}/${route}/${subroute}`, body);
+        const value = await this.agentControllerService.handleRequest(agentId, route, subroute, body);
```
Comment (jsaur) [inline src/controller/agent.controller.controller.ts:27]: Just to make the logs easier to read

File: src/controller/handler/connections.ts [hunk 2]
```diff
@@ -23,7 +23,7 @@ export class Connections implements IAgentResponseHandler {
 
         // if the cacheKey is in the cache then the agent has already accepted the request
         // when we only allow once, there is no need to continue with this message
-        if (this.cache.get<any>(cacheKey) && permissionState === AgentGovernance.PERMISSION_ONCE) {
+        if (await this.cache.get<any>(cacheKey) && permissionState === AgentGovernance.PERMISSION_ONCE) {
```
Comment (jsaur) [inline src/controller/handler/connections.ts:26]: The cache calls needed awaits, and that trickled up to converting various things to async. The issue before was that code was being executed out of sequence due to the lack of awaits

File: src/controller/handler/connections.ts [hunk 3]
```diff
@@ -83,18 +83,15 @@ export class Connections implements IAgentResponseHandler {
 
             Logger.info(`requesting agent to accept connection invite ${req.url}`);
             const res = await this.http.requestWithRetry(req);
-            return res;
+            return res.data;
```
Comment (jsaur) [inline src/controller/handler/connections.ts:86]: I was getting an error converting circular JSON when trying to return the entire res object, instead just returning data

File: src/controller/handler/connections.ts [hunk 4]
```diff
@@ -83,18 +83,15 @@ export class Connections implements IAgentResponseHandler {
...
 
         // this webhook message indicates the receiving agent has accepted the invite and now
         // we need to instruct this agent to finish the steps of a connection
         if (body.state === 'request' && body.initiator === 'self') {
             const action = 'accept-request';
-            this.checkPolicyForAction(action, templatedCacheKey);
-            readPermission(action, templatedCacheKey);
-            // at this point, the other participant in this conversation hasn't caught up with
-            // everything, so we hold our horses for a moment
-            await delay(2000);
```
Comment (jsaur) [inline src/controller/handler/connections.ts:97]: don't need the delay anymore now with the await stuff in

File: test/credential.policies.e2e-spec.ts [hunk 5]
```diff
@@ -137,13 +140,13 @@ describe('Issue and Prove credentials using policies (e2e)', () => {
...
             });
-    }, 30000);
+    });
 
     it ('issuer creates credential definition', async () => {
         await delayFunc(1000);
         const data = {
             schema_id: schemaId,
-            support_revocation: true,
+            support_revocation: false,
```
Comment (jsaur) [inline test/credential.policies.e2e-spec.ts:149]: Setting revocation to true was giving me errors about now revocation registry present

File: test/credential.policies.e2e-spec.ts [hunk 6]
```diff
@@ -216,7 +219,20 @@ describe('Issue and Prove credentials using policies (e2e)', () => {
                     throw e;
                 }
             });
-    }, 30000);
+    });
+
+    it('Holder checks on credential', async () => {
```
Comment (jsaur) [inline test/credential.policies.e2e-spec.ts:224]: Added this test to ensure that the holder actually gets the credential

File: test/credential.policies.e2e-spec.ts [hunk 7]
```diff
@@ -235,18 +251,7 @@ describe('Issue and Prove credentials using policies (e2e)', () => {
                         ]
                     }
                 },
-                requested_predicates: {
```
Comment (jsaur) [inline test/credential.policies.e2e-spec.ts:238]: I couldn't get predicates to work, maybe you can try again

File: test/credential.policies.e2e-spec.ts [hunk 8]
```diff
@@ -277,14 +282,14 @@ describe('Issue and Prove credentials using policies (e2e)', () => {
                 try {
                     Logger.warn(`${agentUrl}/present-proof/records/${presentationExchangeId} result -> ${res.status}`, res.body);
                     expect(res.status).toBe(200);
-                    // TODO: update once an acapy bug is fixed as it doesn't complete this step so the state returned is 'request_sent' not 'verified'
-                    expect(res.body.state).toBe('request_sent');
+                    expect(res.body.state).toBe('verified');
```
Comment (jsaur) [inline test/credential.policies.e2e-spec.ts:285]: Now we're getting the verified state and we can check the verified value too

File: test/credential.policies.e2e-spec.ts [hunk 9]
```diff
@@ -216,7 +219,20 @@ describe('Issue and Prove credentials using policies (e2e)', () => {
                     throw e;
                 }
             });
-    }, 30000);
```
Comment (matt-raffel-kiva) [inline test/credential.policies.e2e-spec.ts:219]: these timeouts we might want to keep.   Both Jeff and I seem to have docker performance issues.  We can add them back later if we need them.....

**Pull request merged**